pennylane_lightning/src/util/TypeTraits.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file
+ * Contains type traits
+ */
+#pragma once
+#include <complex>
+#include <type_traits>
+namespace Pennylane::Util {
+// Enable until C++20 support is explicitly allowed
```

Comment:
If we are bumping GCC and the Ubuntu builder, should we allow this? I assume we may need to revisit MacOs support if so.

Reply:
Yes, we may move all other build systems to GCC>=10. As the latest LTS version of Ubuntu has GCC11 as default (and we need to move C++20 somewhen), I think we need to do this in the not-so-far future. Still, the real problem is `manylinux2014` image we use for `cibuildwheel` (https://github.com/pypa/cibuildwheel/discussions/984).

Reply:
Hmmm, yea, always the issue with the build-wheel. I wonder if it makes sense for us to roll our own wheel-builder based on this image. We already do this in part for LightningGPU, so it wouldn't be too far a stretch to add one with a new toolchain if we need to.

Reply:
I looked for some more information and it seems like moving to GCC10 is not quite difficult. Do you think is it reasonable to go to GCC10 now? Then I will try this.

Reply:
Yea, I think we can go for this, at least for a trial. Maybe create a separate PR so we can compare how the changes influence Win & MacOS builds too.